implementation.js

```diff
@@ -149,28 +161,14 @@ module.exports = function from(arrayLike) {
 		}
 	}
 
-	var len = ES.ToLength(items.length);
+	var arrayLike = ES.ToObject(parseIterableLike(items));
+	var len = ES.ToLength(arrayLike.length);
 	var A = ES.IsCallable(C) ? ES.ToObject(new C(len)) : new Array(len);
```

Comment:
Thanks, this is much better - however, the `new C`/`new Array` call needs to happen before the iteration function happens.

I'm not sure how to achieve this when using `for..of`, but it may not be super important there since if there's no Symbols, there's no observable `[[Get]]`.

Here's the operation order as I read it:
 1. possible TypeError if mapFn is not `== null` and not callable
 1. If Symbols exist, `[[Get]]` of `Symbol.iterator`
 1. if using iterator:
   1. `new C`/`new Array`
   1. if Symbols exist, [invoke](http://www.ecma-international.org/ecma-262/6.0/#sec-getiterator) the `Symbol.iterator` method
 1. else
    1. possible TypeError from ToObject(items)
    1. ToLength(Get(arrayLike, 'length'))
    1. `new C`/`new Array`
 1. the existing while loop